File: 0179-largest-number/0179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        numStrs = list(map(str, nums))
        numStrs.sort(key=cmp_to_key(lambda x, y: 1 if x + y > y + x else -1))


        if numStrs[-1] == '0':
            return '0'
            
        return ''.join(numStrs[::-1])

File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end(), [&] (int const &a, int const &b) {
            return to_string(a) + to_string(b) > to_string(b) + to_string(a);
        });

        if (nums.size() && nums[0] == 0) return "0";

        stringstream ans;
        for (int i = 0; i < nums.size(); i++) ans << to_string(nums[i]);

        return ans.str();
    }
};